### PR TITLE
Media Alignment for text and media on the same row

### PR DESCRIPTION
**Pull request by snschau**
Summary of changes
Note: this replaces PR #153 - will leave it for comments 

Added an option in Media Text Widget to display images on the right of the text block. Left the default (or no choice) to the image displayed on the left.

Frontend
gatsby-node.js

added field_media_alignment to schema for paragraph__media_text type
mediaText.js

added field_media_alignment to the query.
added code to check if field_media_alignment  data exists (default to 'Left' if does not) and assign it to a variable.
refactored code to check what side the image is to appear on and to render it either on the left or right. Alignment choice will if media and text are not in a row.
Backend
Multidev: https://imagealign-chug.pantheonsite.io/

In the Paragraph type: Media and Text added the field Media Alignment - field_media_alignment. Defaults to Left for new items, but existing content with no value will still show on the left if is an image. Not a required field, so when updating existing pages users are not required to change it. But all new content will have a value.

Help text :
 Note: This setting is only when the media (text or video) is displayed beside the text. If other settings place the media above the text this value will be ignored. 

**Left** will place the media to the left of the text.<br>
**Right** will place the media to the right of the text.<br>

Default is Left, on small screens the media will be above the text.

Test Plan
https://sstest.gatsbyjs.io/ is configured to build on this branch and usehttps://imagealign-chug.pantheonsite.io/ as the data source. When testing rebuild SS Test on gatsby cloud or on a local build.

Add a new page and add Media and Text widgets and select Left or Right, add a video to check if this setting is ignored.
Check any existing page that has Media and Text widgets and ensure they are displaying as they should.

https://sstest.gatsbyjs.io/learn/ shows how it will look with alternating left and right images.

https://imagealign-chug.pantheonsite.io/node/876/edit?destination=/admin/content%3Ftitle%3Dlearn%26uid%3D%26type%3DAll%26status%3DAll%26field_tags_target_id_op%3Dor%26field_tags_target_id%3D%26items_per_page%3D50

[Widget Examples](https://sstest.gatsbyjs.io/widget-examples/)
[Media and Text Examples](https://sstest.gatsbyjs.io/media-and-text-examples/)

Both work as expected (with some elements unchanged) note in the section with right columns - I have made the item to have the image on the right, to show that it works in this case).

### CODE REVIEW

File: src/components/shared/mediaText.js [hunk 1]
```diff
@@ -202,7 +203,11 @@ function MediaText (props) {
     }
     headingClass = classNames(headingClass, headingColor);
     textCol = classNames(textCol, textColBg, textColHeight, textColPadding, "text-break");
-    
+
+    if (imageAlignment !== "left") {
+        wrapperCol = wrapperCol + " flex-row-reverse"
```
Comment (tqureshi-uog) [inline src/components/shared/mediaText.js:208]:
```suggestion
        wrapperCol = classNames(wrapperCol, "flex-row-reverse");
```

File: src/components/shared/mediaText.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ function MediaText (props) {
     const imageURL = mediaRelationships?.field_media_image;	
     const imageAlt = props.widgetData?.relationships?.field_media_text_media?.field_media_image?.alt ?? "";
     const mediaSize = props.widgetData?.field_media_image_size;
+    const imageAlignment = props.widgetData?.field_media_alignment ?? 'left';
```
Comment (tqureshi-uog) [inline src/components/shared/mediaText.js:23]:
```suggestion
    const mediaAlignment = props.widgetData?.field_media_alignment ?? 'left';
```

File: src/components/shared/mediaText.js [hunk 3]
```diff
@@ -202,7 +203,11 @@ function MediaText (props) {
     }
     headingClass = classNames(headingClass, headingColor);
     textCol = classNames(textCol, textColBg, textColHeight, textColPadding, "text-break");
-    
+
+    if (imageAlignment !== "left") {
```
Comment (tqureshi-uog) [inline src/components/shared/mediaText.js:207]:
```suggestion
    if (mediaAlignment !== "left") {
```

**Review by tqureshi-uog — CHANGES_REQUESTED** [review]

File: src/components/shared/mediaText.js (hunk 1 above)
Comment (snschau) [reply src/components/shared/mediaText.js:208]: needs the space - otherwise it gets appended to the word before if one is there.

Comment (tqureshi-uog) [reply src/components/shared/mediaText.js]: No, you don't need the space if you use the `classNames` function - see the code for `textCol` on line 204

**Review by tqureshi-uog — APPROVED** [review]

**Review by mmafe — APPROVED** [review]

**Pull request merged**